nexus/social_media/services.py

```diff
@@ -0,0 +1,173 @@
+# Third Party Stuff
+import requests
+from django.conf import settings
+from django.utils import timezone
+from rest_framework import status
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media.models import Post
+
+
+def upload_image_to_linkedin(author, headers, post, api_url_base, linkedin):
+    """Upload image to linkedin for given post.
+
+    :param author: Owner of the post to be used for response POST data.
+
+    :param headers: HTTP headers required for successful HTTP request.
+
+    :param post: Post model instance.
+
+    :param api_url_base: Base url for LinkedIn API.
+
+    :param linkedin: LINKEDIN_AUTH object taken from settings.
+
+    :returns: A valid HTTP response.
+
+    """
+    post_data_assets = {
+        "registerUploadRequest": {
+            "recipes": [
+                "urn:li:digitalmediaRecipe:feedshare-image"
+            ],
+            "owner": author,
+            "serviceRelationships": [
+                {
+                    "relationshipType": "OWNER",
+                    "identifier": "urn:li:userGeneratedContent"
+                }
+            ]
+        }
+    }
+
+    api_url_assets = f"{api_url_base}assets?action=registerUpload"
+
+    response_assets = requests.post(api_url_assets, json=post_data_assets,
+                                    headers=headers)
+    if response_assets.status_code == status.HTTP_200_OK:
+        json_response_assets = response_assets.json()
+        uploadUrl = json_response_assets.get('value')\
+                                        .get('uploadMechanism')\
+                                        .get('com.linkedin.digitalmedia.uploading.MediaUploadHttpRequest')\
+                                        .get('uploadUrl')
+
+        asset = json_response_assets.get('value').get('asset')
+        response_upload_image = requests.post(uploadUrl, data=post.image.file.read(),
+                                              headers={'Authorization': f"Bearer {linkedin['access_token']}"})
+        return (response_upload_image, asset)
+    else:
+        return (response_assets, None)
+
+
+def post_to_linkedin(post_id):
+    """Function to post on linkedin
+
+    :param post_id: UUID of the post instance to be posted.
+
+    :returns: A valid HTTP response
+
+    """
+    linkedin = settings.LINKEDIN_AUTH
+    author = f"urn:li:organization:{linkedin['organization_id']}"
+
+    headers = {'X-Restli-Protocol-Version': '2.0.0',
+               'Content-Type': 'application/json',
+               'Authorization': f"Bearer {linkedin['access_token']}"}
+
+    api_url_base = 'https://api.linkedin.com/v2/'
```

Comment:
This should be in the settings and picked from the settings for this function and all other functions that will use the LinkedIn API.